hazelcast/src/main/java/com/hazelcast/util/JVMUtil.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.util;
+
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeDataSupport;
+import java.lang.management.ManagementFactory;
+
+import static com.hazelcast.logging.Logger.getLogger;
+import static com.hazelcast.nio.UnsafeHelper.UNSAFE;
+import static com.hazelcast.nio.UnsafeHelper.UNSAFE_AVAILABLE;
+
+/**
+ * Helper methods for retrieving JVM specific information.
+ */
+public final class JVMUtil {
+
+    private static final boolean DEFAULT_COMPRESSED_OOPS = true;
+    private static final boolean COMPRESSED_OOPS = isCompressedOopsEnabled();
+    private static final boolean IS_64_BIT = is64Bit();
+
+    public static final int OBJECT_HEADER_IN_BYTES = IS_64_BIT ? (COMPRESSED_OOPS ? 12 : 16) : 8;
+    public static final int REFERENCE_COST_IN_BYTES = COMPRESSED_OOPS ? 4 : 8;
+
+    private JVMUtil() {
+    }
+
+    private static boolean is64Bit() {
+        String systemProp = System.getProperty("com.ibm.vm.bitmode");
+        if (systemProp != null) {
+            return "64".equals(systemProp);
+        }
+
+        systemProp = System.getProperty("sun.arch.data.model");
+        if (systemProp != null) {
+            return "64".equals(systemProp);
+        }
+
+        systemProp = System.getProperty("java.vm.version");
+        return systemProp != null && systemProp.contains("_64");
+    }
+
+    private static boolean isCompressedOopsEnabled() {
+        // Check HotSpot JVM implementation.
+        String enabled = isHotSpotCompressedOopsEnabledOrNull();
+        if (enabled != null) {
+            return Boolean.valueOf(enabled);
+        }
+
+        // Fallback check for other JVM implementations.
+        enabled = isCompressedOopsOrNull();
+        if (enabled != null) {
+            return Boolean.valueOf(enabled);
+        }
+
+        return DEFAULT_COMPRESSED_OOPS;
+    }
+
+    private static String isHotSpotCompressedOopsEnabledOrNull() {
```

Comment:
why are we dealing with `String`? this method (`isHotSpotCompressedOopsEnabledOrNull`) and the other `isCompressedOopsOrNull` could be return boolean.